EventLog.md

```diff
@@ -0,0 +1,397 @@
+    insertObserveCtrl(Ctrl.Pause, sid, why)
+
+  private def insertObserveIntegration(l: Lifecycle, sid: Sequence.Id, step: Int): ConnectionIO[Int] =
+    sql"""
+      INSERT INTO log_observe_int (lifecycle, sequence_id, step)
+           VALUES ($l :: evt_lifecycle,
+                   ${sid.toString},
+                   $step)
+    """.update.run
+
+  def insertStartIntegration(sid: Sequence.Id, step: Int): ConnectionIO[Int] =
+    insertObserveIntegration(Lifecycle.Start, sid, step)
+
+  def insertEndIntegration(sid: Sequence.Id, step: Int): ConnectionIO[Int] =
+    insertObserveIntegration(Lifecycle.End, sid, step)
+
+  private def ts(i: Instant): Timestamp =
+    Timestamp.from(i)
+
+  private def selectAllObserveStage(start: Instant, end: Instant): ConnectionIO[List[EventLog.Event]] =
+    sql"""
+      SELECT timestamp,
+             lifecycle,
+             observe_stage,
+             sequence_id
+        FROM log_observe_stage
+       WHERE timestamp BETWEEN ${ts(start)} AND ${ts(end)}
+    ORDER BY timestamp
+    """.query[(Timestamp, Lifecycle, Stage, String)].map { case (t,l,s,i) =>
+      val timestamp = t.toInstant
+      val sid       = Sequence.Id.unsafeFromString(i)
+      val start     = l == Lifecycle.Start
+
+      s match {
+        case Slew   => start ? (EventStartSlew(timestamp, sid):     Event) | EventEndSlew(timestamp, sid)
+        case Visit  => start ? (EventStartVisit(timestamp, sid):    Event) | EventEndVisit(timestamp, sid)
+        case Seq    => start ? (EventStartSequence(timestamp, sid): Event) | EventEndSequence(timestamp, sid)
+      }
+    }.list
+
+  private def selectAllObserveControl(start: Instant, end: Instant): ConnectionIO[List[EventLog.Event]] =
+    sql"""
+      SELECT timestamp,
+             observe_ctrl,
+             sequence_id,
+             why
+        FROM log_observe_ctrl
+       WHERE timestamp BETWEEN ${ts(start)} AND ${ts(end)}
+    ORDER BY timestamp
+    """.query[(Timestamp, Ctrl, String, Option[String])].map { case (t,c,i,w) =>
+      val timestamp = t.toInstant
+      val sid       = Sequence.Id.unsafeFromString(i)
+
+      c match {
+        case Ctrl.Abort    => EventAbortObserve(   timestamp, sid, w): Event
+        case Ctrl.Continue => EventContinueObserve(timestamp, sid, w): Event
+        case Ctrl.Pause    => EventPauseObserve(   timestamp, sid, w): Event
+        case Ctrl.Stop     => EventStopObserve(    timestamp, sid, w): Event
+      }
+    }.list
+
+  private def selectAllObserveIntegration(start: Instant, end: Instant): ConnectionIO[List[EventLog.Event]] =
+    sql"""
+      SELECT timestamp,
+             lifecycle,
+             sequence_id,
+             step
+        FROM log_observe_int
+       WHERE timestamp BETWEEN ${ts(start)} AND ${ts(end)}
+    ORDER BY timestamp
+    """.query[(Timestamp, Lifecycle, String, Int)].map { case (t,l,i,s) =>
+      val timestamp = t.toInstant
+      val sid       = Sequence.Id.unsafeFromString(i)
```

Comment:
You can add mappings to/from `Instant` and `Sequence.Id` to avoid the extra encoding/decoding here. I have been putting these in the `dao` package object.

```scala
implicit val InstantMeta: Meta[Instant] =
  Meta[Timestamp].nxmap(_.toInstant, Timestamp.from)

implicit val SequencedMeta: Meta[Sequence.Id] =
  Meta[String].nxmap(Sequence.Id.unsafeFromString, _.toString)
```

This will allow you to use these types directly as interpolated values in sql literals and as mapped output types in your results.